gps/identifier.go

```diff
@@ -177,6 +177,8 @@ type bimodalIdentifier struct {
 	prefv Version
 	// Indicates that the bmi came from the root project originally
 	fromRoot bool
+	// The path to the atom in the graph, e.g. root -> foo -> bar
+	path []atom
```

Comment:
OK, so here's where i think the problem is gonna be. This is effectively precaching the path to a given atom from the root - but there's no guarantee that it's the only possible path to the atom. That itself may not _necessarily_ be a problem (i haven't thought it through in at the level of the purest white driven snow algorithm), but it likely will not work well in terms of the way we populate the unselected queue.
 
Have a look at `selectAtom()`. We only add a bmi to the unselected queue if it induces a new package requirement on the target atom. i believe that'll mean that it's possible for a transitive constraint on P to be ignored if project A that expresses it induces no new packages to be selected within P.

That may not be true, but it's not covered by the current set of tests. It also may be the case that this isn't true, but i'm reasoning that it is by relying on cached thoughts around the the restricted purpose of `bimodalIdentifier`. If that's the case, it's evidence for why the reuse of similarly-shaped types for different purposes inside the algorithm is not a great idea - it complicates the already-difficult task of reasoning about the algorithm.